map_conversion.py

```diff
@@ -0,0 +1,206 @@
+                  'singapore-onenorth':'SON_10322', 'singapore-queenstown':'SQT_10324'}
+def populate_lane_dict(nusc_map, root, data):
+    """
+    Return a map that links a nuscenes 'way' token to all the nodes that are a part of that way.
+    Also adds all these nodes to the xml map.
+    """
+    lane_dict = defaultdict(list)
+
+    # Map x,y value of node to the new(argo) node id in the xml
+    present_nodes = {}  # k:tuple, v: node_id
+    global_id = 0  # New id for nodes in the xml
+
+    # Loop over all lanes in the json to populate the lane to node dictionary
+    for way in data['lane']:
+        if way['token'] == '8f23d3ed-1089-4fcf-a178-a174abc69938':
+            continue
+        lane_record = nusc_map.get_lane(way['token'])  # get arcline associated with lane
+        poses = arcline_path_utils.discretize_lane(lane_record,
+                                                   resolution_meters=0.5)  # discretize the lane to given resolution
+        for pose in poses:
+            currNode = (pose[0], pose[1])
+            if currNode not in present_nodes:
+                node = ET.SubElement(root, 'node')
+                node.set('id', str(global_id))
+                present_nodes[currNode] = global_id
+                global_id += 1
+                node.set('x', str(pose[0]))
+                node.set('y', str(pose[1]))
+            lane_dict[way['token']].append(present_nodes[currNode])
+    return lane_dict
+
+def populate_polys(data):
+    """
+    Returns a dictionary poly_dict
+    poly_dict maps a lane polygon in the original map to its exterior points.
+    """
+    # Map nodes in the original json to their x, y coordinates
+    node_dict = {}  # k: node_token v: (x,y)
+
+    # Loop over all nodes in the json to populate the node_dict
+    for node in data['node']:
+        node_dict[node['token']] = (node['x'], node['y'])
+
+    # Map polygons in the original json to the x,y coordinates of their exterior nodes
+    poly_dict = {}  # k: poly_token v: np.ndarray of shape (N,2)
+
+    # Loop over all nodes in the json to populate the node_dict
+    for poly in data['polygon']:
+        poly_array = []
+        for node_id in poly['exterior_node_tokens']:
+            poly_array.append(list(node_dict[node_id]))
+        poly_dict[poly['token']] = np.array(poly_array)
+    return poly_dict
+
+def create_lanes_xml(nusc_map, root, data, filename, argo_dir, lane_dict, poly_dict):
+    """
+    Fill up the xml map file with lane centelines.
+    Also create the supporting files halluc_bbox_table.npy and tableidx_to_laneid_map.json
+    """
+    # Id to assign to lanes in the new xml map file. We arbitrarily start with 8000000.
+    # We make up new lane_ids since the original one's are non numerical
+    global_way_id = 8000000
+
+    # Map that links new lane_id in the xml to its original token in the json.
+    way_to_lane_id = {}  # k: global_way_id , v: data['lane']['token']
```

Comment:
Thanks for the PR, Jeevanjot, looks good.

Can we remove this commented out code here in line 85?

Reply:
Done